css/tokens/index.js

```diff
@@ -1,30 +1,36 @@
-const sassExtract = require('sass-extract');
 const fs = require('fs-extra');
 const path = require('path');
-const { convertSassVariable } = require('./conversion');
 const { quicktype, InputData, jsonInputForTargetLanguage } = require('quicktype-core');
+const { exporter } = require('sass-export');
+const glob = require('fast-glob');
 
 createTokens();
 
 async function createTokens() {
-	const indexFile = require.resolve('../src/tokens/index.scss');
+	const include = './src/tokens/*.scss';
+	const ignore = ['./src/tokens/index.scss'];
+	const filePaths = await glob(include, { ignore });
 
-	/** @type {import('./types').SassConvertRendered }} **/
-	const rendered = await sassExtract.render({
-		file: indexFile
+	filePaths.forEach(path => {
+		try {
+			fs.readFile(path, 'utf8', function read(err, result) {
+				if (err) throw err;
+				if (!result.includes('@sass-export-section')) {
+					console.log(`Warning: ${path} is missing @sass-export-section annotations.`);
+				}
+			});
+		} catch (err) {
+			throw new Error(`Problem reading token files: ${err}`);
+		}
 	});
 
-	const collection = collectTokenSources(rendered);
+	const options = {
+		inputFiles: [require.resolve('../src/tokens/palette.scss'), ...filePaths]
```

Comment:
After giving this a second (and third!) thought, I'm thinking perhaps I steered you in the wrong direction with file-glob. It's definitely true that this globbing works fine, but if need to respect import order, perhaps we should provide a more robust solution to do so always (not just for the one where we need to now). I had a little tinker with this and it seemed to produce the right filePaths in an array and respect import order:

```js
const fs = require('fs-extra');
const path = require('path');

async function getSortOrder(indexPath, filePathStem) {
	try {
		const indexFile = (await fs.readFile(indexPath)).toString();
		const lines = indexFile.split('\n').reduce((arr, line) => {
			console.log(line);  // ⚠ remove me
			if (line.includes('@import')) {
				const filePath = line.replace('@import', '').replace(`'`, '').trim();
				console.log({ filePath });   // ⚠ remove me
				arr.push(path.join(filePathStem, filePath));
			}
			return arr;
		}, []);
	} catch (err) {
		throw err;
	}
}
const filePathStem = path.join(process.cwd(), './css/src/tokens'); // ⚠ this might need changing
const indexPath = path.resolve(filePathStem, 'index.scss'); // ⚠ this might need changing
getSortOrder(indexPath, filePathStem);

```

Reply:
Replaced the previous approach with the function above. Also added a sort function and refactored the comment check into its own function.